src/fsharp/service/ServiceInterfaceStubGenerator.fs

```diff
@@ -213,7 +213,7 @@ module internal InterfaceStubGenerator =
         let nm = 
             match arg.Name with 
             | None ->
-                if arg.Type.HasTypeDefinition && arg.Type.TypeDefinition.XmlDocSig = "T:Microsoft.FSharp.Core.unit" then "()" 
+                if arg.Type.HasTypeDefinition && arg.Type.TypeDefinition.CompiledName = "Microsoft.FSharp.Core.Unit" then "()" 
```

Comment:
Is `Compiled` name really fully qualified at this point?  I thought that would just be `FSharpUnit`?

Reply:
just able to debug it and it's look like it's the CompiledName of unit is unit
![image](https://user-images.githubusercontent.com/1097246/37305355-dcbf7718-263c-11e8-868f-55b5cd0e721f.png)